src/main/java/org/apache/commons/imaging/ImagingParameters.java

```diff
@@ -0,0 +1,333 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Created 2015 by Michael Gross, mgmechanics@mgmechanics.de
+ * Changed 2015 by Michael Gross, mgmechanics@mgmechanics.de
+ */
+
+package org.apache.commons.imaging;
+
+import org.apache.commons.imaging.common.BufferedImageFactory;
+
+/**
+ * This class is a POJO holding data for parameters as requested in IMAGING-159.
+ * It holds data needed for all image formats. For data needed only by one
+ * particular format there are inherited classes.
+ * <p>
+ * There are two kinds of parameters: with and without default value. For
+ * parameters without default value it implements lazy initialization:
+ * <br>
+ * After getting an instance there is no value present. In this state accessing
+ * the value with getX() in this state will cause a RuntimeException.
+ * There is always a isXPresent() method for this parameter which returns
+ * {@code false} in this state.
+ * <br>
+ * Once you provided a value using setX() you may access this value with getX().
+ * The isXPresent() method will return {@code true} now.
+ * <p>
+ * Other parameters have a default value. This is told in the javadoc for their
+ * getX() method. They don't have a isXPresent() method. You may access them any time.
+ * <p>
+ * All boolean parameters have default parameters. Their setter and getter have a different
+ * naming scheme: enableX() causes isXEnabled() to return {@code true} while 
+ * disableX() causes isXEnabled() to return {@code false}.
+ */
+public class ImagingParameters {
+    
+    private boolean verbose; //PARAM_KEY_VERBOSE
+    
+    private boolean strict; // PARAM_KEY_STRICT
+    
+    private String fileNameHint; // PARAM_KEY_FILENAME
+    
+    private String xmpXmlAsString; // PARAM_KEY_XMP_XML
+    
+    private ImageFormat imageFormat; // PARAM_KEY_FORMAT
+    
+    private BufferedImageFactory bufferedImageFactory; // BUFFERED_IMAGE_FACTORY
+    
+    private PixelDensity pixelDensity; // PARAM_KEY_PIXEL_DENSITY
+    
+    /**
+     * This gives you a parameter object with default values.
+     */
+    public ImagingParameters() {
+        this.verbose = false;
+        this.strict = false;
+        this.fileNameHint = null;
+        this.xmpXmlAsString = null;
+        this.imageFormat = null;
+        this.bufferedImageFactory = null;
+        this.pixelDensity = null;
+    }
+    
+    //****** verbose ******
+    /**
+     * Tells if the verbose mode is turned on/off.
+     * <p>
+     * Parameter applies to read and write operations.
+     * @return Valid values: {@code true} means {@literal "verbose mode enabled"}
+     * and {@code false} means {@literal "verbose mode disabled"} (default value).
+     */
+    public boolean isVerboseEnabled() {
+        return this.verbose;
+    }
+    
+    /**
+     * Turns the verbose mode on.
+     * Parameter applies to read and write operations.
+     */
+    public void enableVerbose() {
```

Comment:
On 23.02.2015 19:59, Benedikt Ritter
      wrote:
    
    
      In src/main/java/org/apache/commons/imaging/ImagingParameters.java:
      > +    /**
> +     * Tells if the verbose mode is turned on/off.
> +     * <p>
> +     * Parameter applies to read and write operations.
> +     * @return Valid values: {@code true} means {@literal "verbose mode enabled"}
> +     * and {@code false} means {@literal "verbose mode disabled"} (default value).
> +     */
> +    public boolean isVerboseEnabled() {
> +        return this.verbose;
> +    }
> +    
> +    /**
> +     * Turns the verbose mode on.
> +     * Parameter applies to read and write operations.
> +     */
> +    public void enableVerbose() {

      Would be better to add a plain old setter method so that this
        class can be used as a JavaBean.
      —
        Reply to this email directly or view
          it on GitHub.
      
        
          
          
        
        
      
    
    > so that this class can be used as a JavaBean
    
    Yesterday I wrote "You already can! All boolean parameters have a
    default value. So they are initialized when the class is
    instanziated.". I was a little bit fast.  Of course, if we need a
    full JavaBean we need to have plain old setter methods even for
    boolean values (for values of all other types there is one). So I
    just implemented plain setters.
    
    What is the intended use case for a JavaBean here? 
    -- 
Michael Groß